File: MiniFTPServer/LibFTPServ/HelperFunctions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Security.Cryptography;

namespace LibFTPServ
{
    internal static class HelperFunctions
    {
        public static string FixPath(string path)
        {
            StringBuilder sb = new StringBuilder();
            sb.Append("/");
            string[] p = path.Split('/');
            for (int i = 0; i < p.Length; i++)
            {
                if (!string.IsNullOrEmpty(p[i]))
                {
                    sb.Append(p[i]);
                    if (i != p.Length - 1) sb.Append("/");
                }
            }
            return sb.ToString();

        }

        public static string SHA1Hash(string source)
        {
            SHA1CryptoServiceProvider sha1 = new SHA1CryptoServiceProvider();
            byte[] data = Encoding.UTF8.GetBytes(source);
            byte[] hash = sha1.ComputeHash(data);
            StringBuilder sb = new StringBuilder(50);
            for (int i = 0; i < hash.Length; i++) sb.Append(hash[i].ToString("x2").ToLower());
            return sb.ToString(); ;
        }

        public static string GenerateUniqueFileName()
        {
            RNGCryptoServiceProvider rgn = new RNGCryptoServiceProvider();
            byte[] data = new byte[16];
            rgn.GetBytes(data);
            Guid name = new Guid(data);
            return name.ToString();
        }
    }
}

File: MiniFTPServer/vb/LibFTPServ.ConvertedToVBNet/HelperFunctions.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Security.Cryptography

Friend NotInheritable Class HelperFunctions
	Private Sub New()
	End Sub
	Public Shared Function FixPath(path As String) As String
		Dim sb As New StringBuilder()
		sb.Append("/")
		Dim p As String() = path.Split("/"C)
		For i As Integer = 0 To p.Length - 1
			If Not String.IsNullOrEmpty(p(i)) Then
				sb.Append(p(i))
				If i <> p.Length - 1 Then
					sb.Append("/")
				End If
			End If
		Next
		Return sb.ToString()

	End Function

	Public Shared Function SHA1Hash(source As String) As String
		Dim sha1 As New SHA1CryptoServiceProvider()
		Dim data As Byte() = Encoding.UTF8.GetBytes(source)
		Dim hash As Byte() = sha1.ComputeHash(data)
		Dim sb As New StringBuilder(50)
		For i As Integer = 0 To hash.Length - 1
			sb.Append(hash(i).ToString("x2").ToLower())
		Next
		Return sb.ToString()
		

	End Function

	Public Shared Function GenerateUniqueFileName() As String
		Dim rgn As New RNGCryptoServiceProvider()
		Dim data As Byte() = New Byte(15) {}
		rgn.GetBytes(data)
		Dim name As New Guid(data)
		Return name.ToString()
	End Function
End Class
